srv/src/test/java/my/bookshop/GenreHierarchyTest.java

```diff
@@ -113,8 +140,108 @@ void testExpandAll() throws Exception {
-		if (env.acceptsProfiles(Profiles.of("hybrid"))) {
-			expectactions.andExpect(jsonPath("$.value[0].LimitedDescendantCount").value(9));
-		}	
+		if (isOnHana()) {
+			expectations.andExpect(jsonPath("$.value[0].LimitedDescendantCount").value(9));
+		}
+	}
+
+	@Test
+	@WithMockUser(username = "admin")
+	void testSearch() throws Exception {
+		ResultActions expectations = client.perform(get(genresURI
+				+ "?$select=DistanceFromRoot,DrillState,ID,LimitedDescendantCount,name"
+				+ "&$apply=ancestors($root/GenreHierarchy,GenreHierarchy,ID,search(\"ry\"),keep start)"
+				+ "/orderby(name)"
+				+ "/com.sap.vocabularies.Hierarchy.v1.TopLevels(HierarchyNodes=$root/GenreHierarchy,HierarchyQualifier='GenreHierarchy',NodeProperty='ID')"
+				+ "&$count=true"))
+				.andExpect(status().isOk())
+				.andExpect(jsonPath("$.value[0].ID").value(10))
+				.andExpect(jsonPath("$.value[0].name").value("Fiction"))
+				.andExpect(jsonPath("$.value[0].DrillState").value("expanded"))
+				.andExpect(jsonPath("$.value[0].DistanceFromRoot").value(0))
+				.andExpect(jsonPath("$.value[1].ID").value(19))
+				.andExpect(jsonPath("$.value[1].name").value("Fairy Tale"))
+				.andExpect(jsonPath("$.value[1].DrillState").value("leaf"))
+				.andExpect(jsonPath("$.value[1].DistanceFromRoot").value(1))
+				.andExpect(jsonPath("$.value[2].ID").value(16))
+				.andExpect(jsonPath("$.value[2].name").value("Mystery"))
+				.andExpect(jsonPath("$.value[2].DrillState").value("leaf"))
+				.andExpect(jsonPath("$.value[2].DistanceFromRoot").value(1))
+				.andExpect(jsonPath("$.value[3].ID").value(12))
+				.andExpect(jsonPath("$.value[3].name").value("Poetry"))
+				.andExpect(jsonPath("$.value[3].DrillState").value("leaf"))
+				.andExpect(jsonPath("$.value[3].DistanceFromRoot").value(1))
+				.andExpect(jsonPath("$.value[4]").doesNotExist());
+		if (isOnHana()) {
+			expectations.andExpect(jsonPath("$.value[0].LimitedDescendantCount").value(3))
+					.andExpect(jsonPath("$.value[1].LimitedDescendantCount").value(0))
+					.andExpect(jsonPath("$.value[2].LimitedDescendantCount").value(0))
+					.andExpect(jsonPath("$.value[3].LimitedDescendantCount").value(0));
+		}
+	}
+
+	@Test
+	@WithMockUser(username = "admin")
+	void testFilterNotExpanded() throws Exception {
+		client.perform(get(genresURI
+				+ "?$select=DrillState,ID,name,DistanceFromRoot"
+				+ "&$apply=ancestors($root/GenreHierarchy,GenreHierarchy,ID,filter(name eq 'Autobiography'),keep start)/orderby(name)"
+				+ "/com.sap.vocabularies.Hierarchy.v1.TopLevels(HierarchyNodes=$root/GenreHierarchy,HierarchyQualifier='GenreHierarchy',NodeProperty='ID',Levels=1)"))
+				.andExpect(status().isOk())
+				.andExpect(jsonPath("$.value[0].ID").value(20))
+				.andExpect(jsonPath("$.value[0].name").value("Non-Fiction"))
+				.andExpect(jsonPath("$.value[0].DrillState").value("collapsed"))
+				.andExpect(jsonPath("$.value[0].DistanceFromRoot").value(0))
+				.andExpect(jsonPath("$.value[1]").doesNotExist());
+	}
+
+	@Test
```

Comment:
This test currently is red, when running on HANA. The fix will come only in 3.6.1.
I would ignore it so far. What do you think?

Reply:
fine with me